test/runtests.jl

```diff
@@ -321,3 +321,38 @@ end
     @test isapprox(lm(Xf, y).pp.beta0, ones(2))
     @test isapprox(lm(X, yf).pp.beta0, ones(2))
 end
+
+@testset "F test for model comparison" begin
+    d = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
+                  Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2],
+                  Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1])
+    mod = lm(@formula(Result~Treatment), d).model
+    othermod = lm(@formula(Result~Other), d).model
+    nullmod = lm(@formula(Result~1), d).model
+    @test GLM.issubmodel(nullmod, mod)
+    @test !GLM.issubmodel(othermod, mod)
+    
+    @test_throws ArgumentError ftest(mod, othermod)
+    
+    ft = ftest(mod, nullmod)
+    @test isapprox(ft.pval[1].v,  2.481215056713184e-8)
+    # Test output
+    @test sprint(show, ftest(mod, nullmod)) == 
+        """
+                Res. DOF DOF    ΔDOF    SSR    ΔSSR    R²      ΔR²    F*       p(>F) 
+        Model 1 10.0000  3.0000         0.1283         0.9603                        
```

Comment:
Not a big deal, but I wonder whether it would be possible to force `@printf` to align number on the decimal separator (see e.g. the R² column).